src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,299 @@
+    }
+  }
+
+  // Find from node_to_created_actors_.
+  auto iter = node_to_created_actors_.find(node_id);
+  if (iter != node_to_created_actors_.end()) {
+    auto created_actors = std::move(iter->second);
+    // Remove all created actors from node_to_created_actors_.
+    node_to_created_actors_.erase(iter);
+    for (auto &entry : created_actors) {
+      // Remove the actor from worker_to_created_actor_.
+      worker_to_created_actor_.erase(entry.second->GetWorkerID());
+      // Reconstruct the removed actor.
+      ReconstructActor(entry.second);
+    }
+  }
+}
+
+void GcsActorManager::ReconstructActor(std::shared_ptr<GcsActor> actor,
+                                       bool need_reschedule) {
+  RAY_CHECK(actor != nullptr);
+  auto node_id = actor->GetNodeID();
+  auto worker_id = actor->GetWorkerID();
+  RAY_LOG(WARNING) << "Actor is failed " << actor->GetActorID() << " on worker "
+                   << worker_id << " at node " << node_id
+                   << ", need_reschedule = " << need_reschedule;
+  actor->UpdateAddress(rpc::Address());
+  auto mutable_actor_table_data = actor->GetMutableActorTableData();
+  // If the need_reschedule is set to false, then set the `remaining_reconstructions` to 0
+  // so that the actor will never be rescheduled.
+  auto remaining_reconstructions =
+      need_reschedule ? mutable_actor_table_data->remaining_reconstructions() : 0;
+  if (remaining_reconstructions > 0) {
+    mutable_actor_table_data->set_remaining_reconstructions(--remaining_reconstructions);
+    mutable_actor_table_data->set_state(rpc::ActorTableData::RECONSTRUCTING);
+    auto actor_table_data =
+        std::make_shared<rpc::ActorTableData>(*mutable_actor_table_data);
+    // The backend storage is reliable in the future, so the status must be ok.
+    RAY_CHECK_OK(actor_info_accessor_.AsyncUpdate(actor->GetActorID(), actor_table_data,
+                                                  [this, actor](Status status) {
+                                                    RAY_CHECK_OK(status);
+                                                    gcs_actor_scheduler_->Schedule(actor);
+                                                  }));
+  } else {
+    mutable_actor_table_data->set_state(rpc::ActorTableData::DEAD);
+    auto actor_table_data =
+        std::make_shared<rpc::ActorTableData>(*mutable_actor_table_data);
+    // The backend storage is reliable in the future, so the status must be ok.
+    RAY_CHECK_OK(
+        actor_info_accessor_.AsyncUpdate(actor->GetActorID(), actor_table_data, nullptr));
+  }
+}
+
+void GcsActorManager::SchedulePendingActors() {
+  if (pending_actors_.empty()) {
+    return;
+  }
+
+  RAY_LOG(DEBUG) << "Scheduling actor creation tasks, size = " << pending_actors_.size();
+  // TODO(ZhuSenlin): This loop maybe inefficient, it will be optimized later.
+  auto actors = std::move(pending_actors_);
+  while (!actors.empty()) {
+    auto actor = actors.front();
+    actors.pop();
```

Comment:
Nit, I don't think you need to pop here, you can just iterate.

Reply:
> Nit, I don't think you need to pop here, you can just iterate.

The `actors` is a std::queue, which cannot be iterated.